WordPress/gutenberg
# Cover: Set custom color when applying initial background image

## Description

<img width="632" alt="Screen Shot 2020-05-22 at 5 27 47 PM" src="https://user-images.githubusercontent.com/2322354/82711217-bb30a480-9c52-11ea-8a54-269ed60167f4.png">

This update adds an enhancement to the Cover block for background images. When an image is first set, it extracts and the primary color of the image and applies it as a custom background color.

This is achieved by using a library [color-thief](https://github.com/lokesh/color-thief) ([2.6 kb gzipped](https://bundlephobia.com/result?p=colorthief@2.3.0)), which leverages `<Canvas />` for image parsing and color picking.

Here is a video demo of it working:
https://d.pr/v/f2flil

Here's the difference:

<img width="651" alt="Screen Shot 2020-05-22 at 5 44 42 PM" src="https://user-images.githubusercontent.com/2322354/82711645-f4b5df80-9c53-11ea-80e1-558c7f8ed486.png">


## How has this been tested?

Tested locally in Gutenberg

* Run `npm install`
* Run `npm run dev`
* Add a Cover block
* Add an image from the Media Library for the background (perhaps something colourful)
* Check to see if the background tint matches the image
* Check the Cover background color - make sure it's custom

## Types of changes

* This update adds a new `useColorExtract` hook from `@wordpress/components`, which handles the responsibility for image rendering/color extraction

## Checklist:

Resolves comment in:
https://github.com/WordPress/gutenberg/issues/16479#issuecomment-546547459
